File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        #HashMap< , > hm = new HashMap<>();
        dict = { }
        n = len(nums)
        for num in nums:
            if num in dict:
                dict[num] += 1
            else:
                dict[num] = 1 
            

        temp = n/2
        #max_num = 0

        for key,value in dict.items():
            if value > temp:
                #max_num = key
                return key

        return -1
        
        

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> hm = new HashMap<>();
        for(int num : nums){
            if(hm.containsKey(num)){
                hm.put(num,hm.getOrDefault(num,0)+1);
            }
            else{
                hm.put(num,1);
            }
        }

        int temp = nums.length/2;
        /*for(int key: hm.keySet()){
            if(hm.get(key) > temp){
                return key;
            }
        }*/
        for(Map.Entry<Integer,Integer> entry : hm.entrySet()){
            if(entry.getValue()>temp){
                return entry.getKey();
            }
        }
        return -1;
    }
}